packages/loot-core/src/server/main.js

```diff
@@ -986,68 +991,97 @@ handlers['account-move'] = mutator(async function ({ id, targetId }) {
 
 let stopPolling = false;
 
-handlers['poll-web-token'] = async function ({ token }) {
-  let [[, userId], [, key]] = await asyncStorage.multiGet([
-    'user-id',
-    'user-key',
-  ]);
+handlers['poll-web-token'] = async function ({
+  upgradingAccountId,
+  requisitionId,
+}) {
+  let userToken = await asyncStorage.getItem('user-token');
 
-  let startTime = Date.now();
-  stopPolling = false;
+  if (userToken) {
+    let startTime = Date.now();
+    stopPolling = false;
 
-  async function getData(cb) {
-    if (stopPolling) {
-      return;
-    }
+    async function getData(cb) {
+      if (stopPolling) {
+        return;
+      }
 
-    if (Date.now() - startTime >= 1000 * 60 * 10) {
-      cb('timeout');
-      return;
-    }
+      if (Date.now() - startTime >= 1000 * 60 * 10) {
+        cb('timeout');
+        return;
+      }
 
-    let data = await post(
-      getServer().PLAID_SERVER + '/get-web-token-contents',
-      {
-        userId,
-        key,
-        token,
-      },
-    );
+      let data = await post(
+        getServer().NORDIGEN_SERVER + '/get-accounts',
+        {
+          upgradingAccountId,
+          requisitionId,
+        },
+        {
+          'X-ACTUAL-TOKEN': userToken,
+        },
+      );
 
-    if (data) {
-      if (data.error) {
-        cb('unknown');
+      if (data) {
+        if (data.error) {
+          cb('unknown');
+        } else {
+          cb(null, data);
+        }
       } else {
-        cb(null, data);
+        setTimeout(() => getData(cb), 3000);
       }
-    } else {
-      setTimeout(() => getData(cb), 3000);
     }
-  }
 
-  return new Promise(resolve => {
-    getData((error, data) => {
-      if (error) {
-        resolve({ error });
-      } else {
-        resolve({ data });
-      }
+    return new Promise(resolve => {
+      getData((error, data) => {
+        if (error) {
+          resolve({ error });
+        } else {
+          resolve({ data });
+        }
+      });
     });
-  });
+  }
+
+  return null;
 };
 
 handlers['poll-web-token-stop'] = async function () {
   stopPolling = true;
   return 'ok';
 };
 
+handlers['create-web-token'] = async function ({
+  upgradingAccountId,
+  institutionId,
+  accessValidForDays,
+}) {
+  let userToken = await asyncStorage.getItem('user-token');
+
+  if (userToken) {
+    return await post(
+      getServer().NORDIGEN_SERVER + '/create-web-token',
+      {
+        upgradingAccountId,
+        institutionId,
+        accessValidForDays,
+      },
+      {
+        'X-ACTUAL-TOKEN': userToken,
+      },
+    );
+  }
+  return null;
```

Comment:
Currently this causes the call in the frontend to fail because it tries to destructure null.

```suggestion
  return { error: 'unauthorized' };
```